native/src/main/java/io/ballerina/lib/aws/mpe/CommonUtils.java

```diff
@@ -0,0 +1,181 @@
+            }
+        }
+        return requestBuilder.build();
+    }
+
+    public static BMap<BString, Object> getBallerinaResponse(GetEntitlementsResponse nativeResponse) {
+        BMap<BString, Object> entitlementsResponse = ValueCreator.createRecordValue(
+                ModuleUtils.getModule(), Constants.MPE_RESPONSE);
+        String nextToken = nativeResponse.nextToken();
+        if (Objects.nonNull(nextToken)) {
+            entitlementsResponse.put(Constants.MPE_RESPONSE_NXT_TOKEN, StringUtils.fromString(nextToken));
+        }
+        BArray bEntitlements = ValueCreator.createArrayValue(ENTITLEMENT_ARR_TYPE);
+        nativeResponse.entitlements().forEach(e -> {
+            BMap<BString, Object> bEntitlement = convertToBEntitlement(e);
+            bEntitlements.append(bEntitlement);
+        });
+        entitlementsResponse.put(Constants.MPE_RESPONSE_ENTITLEMENTS, bEntitlements);
+        return entitlementsResponse;
+    }
+
+    private static BMap<BString, Object> convertToBEntitlement(Entitlement nativeEntitlement) {
+        BMap<BString, Object> bEntitlement = ValueCreator.createRecordValue(ENTITLEMENT_REC_TYPE);
+        String productCode = nativeEntitlement.productCode();
+        if (Objects.nonNull(productCode)) {
+            bEntitlement.put(Constants.MPE_ENTITLEMENT_PRODUCT_CODE, StringUtils.fromString(productCode));
+        }
+        String dimension = nativeEntitlement.dimension();
+        if (Objects.nonNull(dimension)) {
+            bEntitlement.put(Constants.MPE_ENTITLEMENT_DIMENSION, StringUtils.fromString(dimension));
+        }
+        String customerIdentifier = nativeEntitlement.customerIdentifier();
+        if (Objects.nonNull(dimension)) {
+            bEntitlement.put(Constants.MPE_ENTITLEMENT_CUS_IDNFR, StringUtils.fromString(customerIdentifier));
+        }
+        Instant expirationDate = nativeEntitlement.expirationDate();
+        if (Objects.nonNull(expirationDate)) {
+            long epochSecond = expirationDate.getEpochSecond();
+            int nanos = expirationDate.getNano();
+            BArray utcTime = ValueCreator.createTupleValue(BUTC_TYPE);
+            utcTime.append(epochSecond);
+            utcTime.append(nanos);
+            bEntitlement.put(Constants.MPE_ENTITLEMENT_EXP_DATE, utcTime);
+        }
+        populateEntitlementValue(nativeEntitlement, bEntitlement);
+        return bEntitlement;
+    }
+
+    private static void populateEntitlementValue(Entitlement nativeEntitlement, BMap<BString, Object> bEntitlement) {
+        EntitlementValue nativeEntitlementValue = nativeEntitlement.value();
+        Boolean entitlementBoolVal = nativeEntitlementValue.booleanValue();
+        if (Objects.nonNull(entitlementBoolVal)) {
+            bEntitlement.put(Constants.MPE_ENTITLEMENT_VALUE, entitlementBoolVal);
+            return;
+        }
+        Double entitlementDoubleVal = nativeEntitlementValue.doubleValue();
```

Comment:
Is it correct to use `doubleValue()` for a field having the `decimal` type? I think `Double` type in java use only 64 bits to represent a number while `decimal` can have 128 bits.
https://ballerina.io/spec/lang/master/#section_5.2.4.2

Reply:
Fixed with [94b75ef006413d04d3bbb7845f6e79fe967158e5](https://github.com/ballerina-platform/module-ballerinax-aws.marketplace.mpe/pull/2/commits/94b75ef006413d04d3bbb7845f6e79fe967158e5)